tm/src/main/java/io/seata/tm/TransactionManagerHolder.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.tm;
+
+import io.seata.common.loader.EnhancedServiceLoader;
+import io.seata.core.exception.TransactionException;
+import io.seata.core.exception.TransactionExceptionCode;
+import io.seata.core.model.GlobalStatus;
+import io.seata.core.model.TransactionManager;
+import io.seata.core.protocol.transaction.*;
+import io.seata.core.rpc.netty.TmRpcClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeoutException;
+
+/**
+ * The type Default transaction manager.
+ *
+ * @author sharajava
+ */
+public class TransactionManagerHolder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransactionManagerHolder.class);
+
+    private static class SingletonHolder {
+
+        private static TransactionManager INSTANCE = null;
+
+        static {
+            try {
+                INSTANCE = EnhancedServiceLoader.load(TransactionManager.class);
+                LOGGER.info("TransactionManager Singleton " + INSTANCE);
+            } catch (Throwable anyEx) {
+                LOGGER.error("Failed to load TransactionManager Singleton! ", anyEx);
```

Comment:
should throw exx?

Reply:
Not sure, I was just considering that exception thrown from static block will cause some error of class initialization. 

Reference: https://stackoverflow.com/questions/2070293/why-doesnt-java-allow-to-throw-a-checked-exception-from-static-initialization-b

Reply:
maybe catch check exception EnhancedServiceNotFoundException. other excepiton should throw?